src/coord/cartesian/Cartesian2D.ts

```diff
@@ -146,15 +146,15 @@ class Cartesian2D extends Cartesian<Axis2D> implements CoordinateSystem {
         return out;
     }
 
-    pointToData(point: number[], out?: number[]): number[] {
+    pointToData(point: number[], out?: number[], reserved?: number[], clamp?: boolean): number[] {
```

Comment:
This `reserved` parameter in the base class is preserved for parameters like `clamp`. So there is no need to add another clamp parameter.  Also, the order is wrong. It should be `pointToData(point, clamp, out)`;

There are two other changes I think we can do to simplify the design after searched the code.
1. Remove the `out` parameter because I found there are no lines of code use it.
2. Replace the `reserved?: any` parameter with a `clamp?: boolean` parameter because I didn't see any other possibilities. Correct me if I'm wrong

Reply:
@pissang But, in base class [`CoordinateSystem`](https://github.com/apache/echarts/blob/master/src/coord/CoordinateSystem.ts#L133), the params for `pointToData` are ` point: number[],  reserved?: any, out?: number[]`, I have tried rearrange to ` point: number[],  out?: number[], reserved?: any`, but that would break type.